ui/src/routes/organizations/$orgId/infrastructure-services/create/index.tsx

```diff
@@ -218,18 +215,27 @@ const CreateInfrastructureServicePage = () => {
                       </SelectTrigger>
                     </FormControl>
                     <SelectContent>
-                      {secrets?.data.map((secret) => (
-                        <SelectItem key={secret.name} value={secret.name}>
-                          {secret.name}
-                        </SelectItem>
-                      ))}
+                      {secrets.map((secret) => {
+                        const hierarchyLabel =
+                          secret.hierarchy.charAt(0).toUpperCase() +
+                          secret.hierarchy.slice(1);
```

Comment:
How about extracting this to a `capitalize()` (or similar) helper function for reuse, covered with a few tests? As part of testing, we should also define how to behave for mixed-case strings like "fOO"... I guess we show lowercase the tail, so the final string would be "Foo".